sql/src/main/java/org/apache/druid/sql/calcite/rule/DruidLogicalValuesRule.java

```diff
@@ -128,6 +128,11 @@ static Object getValueFromLiteral(RexLiteral literal, PlannerContext plannerCont
       case TIMESTAMP:
       case DATE:
         return Calcites.calciteDateTimeLiteralToJoda(literal, plannerContext.getTimeZone()).getMillis();
+      case NULL:
```

Comment:
Could this have been the case when Boolean is of null type? 
Referring to this conversation: https://github.com/apache/druid/pull/11923#discussion_r766485610

Reply:
If it was, wouldn't we see an unsupported type exception? do you mean something different

Reply:
It seems I mixed up `literal.getType().getSqlTypeName()` and `literal.getSqlTypeName()` when I originally wanted to ask the question. 

In the current modification, instead of returning `null`, should it be `throw new UnsupportedSQLQueryException("%s type is not supported", literal.getType().getSqlTypeName());`, to preserve the original behavior.

Reply:
that's what it was doing earlier which means we can't support null literals in SQL queries.

Reply:
this change fixes that exception.

Reply:
Got it, thanks. I was incorrectly assuming that this was aimed at making the error message more specific to null columns.